### PR TITLE
8999-Assignment ui change assigned

### PR DESCRIPTION
**Pull request by jimmyPasta**

### CODE REVIEW

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs [hunk 1]
```diff
@@ -161,25 +163,85 @@ public async Task AddAssignee(Guid[] applicationIds, string AssigneeKeycloakId,
...
+                    var application = await _applicationRepository.GetAsync(applicationId);
+                    IQueryable<ApplicationUserAssignment> queryableAssignment = _userAssignmentRepository.GetQueryableAsync().Result;
+                    var assignments = queryableAssignment.Where(a => a.ApplicationId.Equals(applicationId)).Where(b => b.OidcSub.Equals(AssigneeKeycloakId)).ToList();
+                    // Only remove the assignee if they were already assigned
+                    if (application != null && assignments != null)
+                    {
+                        await _userAssignmentRepository.DeleteAsync(assignments.FirstOrDefault());
+                    }
+                }
+                catch (Exception ex)
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs:202]: Can remove the Exception catch block here, ABP has ways of handling the exceptions. 
Throw exception explicitly from here if needed and are created and inherited from the ABP exceptions
Could throw a EntityNotFoundException or AbpValidationException based on Validation logic here

**Review by AndreGAot — CHANGES_REQUESTED** [review]
Exception handling is configure and handled by ABP unless we want to be explicit.
We could add some Unit Tests for Adding and removing Assignees with this PR.

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs [hunk 2]
```diff
@@ -161,25 +168,78 @@ public async Task AddAssignee(Guid[] applicationIds, string AssigneeKeycloakId,
                 try
                 {
                     var application = await _applicationRepository.GetAsync(applicationId);
-                    if (application != null)
+                    var assignees = getAssignees(applicationId);
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs:171]: Think this was like this before but rather PascalCase for method names, and ideally private GetAssigneesAsync which should use async await and not .Result

**Review by AndreGAot — CHANGES_REQUESTED** [review]

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs [hunk 3]
```diff
@@ -161,25 +168,78 @@ public async Task AddAssignee(Guid[] applicationIds, string AssigneeKeycloakId,
...
+                var assignments = queryableAssignment.Where(a => a.ApplicationId.Equals(applicationId)).Where(b => b.OidcSub.Equals(AssigneeKeycloakId)).ToList();
+                // Only remove the assignee if they were already assigned
+                if (application != null && assignments != null)
+                {
+                    await _userAssignmentRepository.DeleteAsync(assignments.FirstOrDefault());
+                }
+            }
+        }
 
+        public async Task ModifyAssignees(dynamic modifiedAssignees)
```
[the remark this answers is not in the document]
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs:207]: Yes, I think it needs to be dynamic, will rename

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs [hunk 4]
```diff
@@ -161,25 +168,78 @@ public async Task AddAssignee(Guid[] applicationIds, string AssigneeKeycloakId,
...
                     }
                 }
                 catch (Exception ex)
                 {
                     Debug.WriteLine(ex.ToString());
                 }
+            }
+        }
+
+        public async Task RemoveAssignee(Guid[] applicationIds, string AssigneeKeycloakId)
```
[the remark this answers is not in the document]
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs]: Renamed to DeleteAssigneeAsync

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs (hunk 2 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs]: Renamed to GetAssigneesAsync and refactored surounding code

**Review by AndreGAot — APPROVED** [review]
Approved

**Pull request merged**